### ISSUE TITLE
import/newline-after-import:  allow disable newline after required in block

### ISSUE THREAD

**Issue by ntnyq** [import-js/eslint-plugin-import#2458]
Expected:

```ts
import { foo } from 'bar'
// newline required here

declare module '*.foo' {
  import { type Foo } from 'bar'
  // no newline required here
  const content: Foo<{}>
  export default content
}
```

An option like `topLevelOnly` or `disabledInBlockScope` would be nice.

**Comment by JounQin** [comment]
I don't quite understand why this is useful, what's the purpose.

**Comment by ljharb** [comment]
You should be able to enforce the same rules in a `declare module` block as you do at the top level of a file.

**Comment by JounQin** [comment]
> You should be able to enforce the same rules in a `declare module` block as you do at the top level of a file.

This issue is asking a new option to `disable` this behavior.

**Comment by ljharb** [comment]
To be able to write the top-level file in one style, and a declare module block inside that file in a different style.

**Comment by JounQin** [comment]
> To be able to write the top-level file in one style, and a declare module block inside that file in a different style.

But why should different styles be used. 😂

**Comment by JounQin** [comment]
> the same reason most styles are chosen

They often only apply one style?

**Comment by ljharb** [comment]
Often, yes - but it's never an inherent restriction. Using eslint overrides, you can have N files each with their own distinct style if you want. While an eslint plugin maintainer _should_ use their subjective bias to decide what styles are supported, I think they should _not_ generally decide which "regions of code" get a style or do not.

In the case of `declare module`, it's akin to a separate file. Separate files should be able to have their own unique style applied via config.

**Comment by JounQin** [comment]
Well, I can't agree with that, but whatever. 🤣
